pkg/tls/config.go

```diff
@@ -3,25 +3,22 @@ package tls
 import (
 	"crypto/tls"
 	"fmt"
+	"sort"
 	"strings"
-
-	"github.com/rancher/rancher/pkg/settings"
```

Comment:
I did mention in the commit message that this is a refactoring of this part of the package.

There were a few issues before:
- unused package-level constants
- needless creation of a list of allowed versions/ciphers even when there is no validation error
- no pre-allocated slices
- not very good names
- coupling between this tls package and settings (these functions shouldn't care where the values are coming from)

I added tests first, then re-wrote this file. I agree that GH displays a lot of changes, and it might be hard to follow, but this is pretty much a rewrite of the file. Perhaps it's best to just review the new file on its own, without looking at how it was before.

Reply:
> I did mention in the commit message that this is a refactoring of this part of the package.

Yes, but you didn't say why. Could you put all those reasons in your commit message?

And better yet, could you separate the refactoring into a separate commit? This is widely considered best practice:

> In practice it often comes down to not mixing refactorings with new features. There are a few reasons for that:

> By mixing both things you increase the chance of having conflicts which are hard to resolve – not only for your code but also for code which is developed in parallel. Your code becomes harder to review: A Reviewer has to think about if a change is related to the new feature or just a rewrite of the existing code. Reverting your changes is becoming harder. Two atomic commits are much better: If there is an issue with the refactoring – Just revert it. If there is an issue with your new feature – Just revert it! A commit indicating “just did some cosmetic changes” might be a good hint to look further in the history.

https://programmerfriend.com/git-best-practices/

Reply:
Also agree. Here, I thought it's just harder to rewrite, refactor, add new logic in separate commits. Normally, of course, I'd have separate commits for refactoring and tests.
I outlined the main reasons for the refactor above, just didn't include them in the commit message, because the overall change is small enough. And what would say? Not well-written functions?

Reply:
Could you just include those reasons in the commit message, that way someone wondering why it changed doesn't have to go to github and find this particular comment thread.